prow/config/config_fuzz.go

```diff
@@ -0,0 +1,26 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import fuzz "github.com/google/gofuzz"
+
+func FuzzParseProwConfig(data []byte) int {
+	var c Config
+	fuzz.NewFromGoFuzz(data).Fuzz(&c)
```

Comment:
random value in prow config? have you tried it?

Reply:
My understanding is that this generates a valid Prow config with random inputs for each field. I haven't tried it yet. test-infra fuzz job is not doing anything important rn so I thought it would be easiest to test in prod

Reply:
Why is this in the production code and unused and not a test? And what is it supposed to test, that we don't panic?

Reply:
I'm experimenting with [ClusterfuzzLite](https://google.github.io/clusterfuzzlite/) to do some continuous fuzz testing. I am not very experienced with writing fuzz tests so this is just my first attempt. My understanding is that It should test that it does not panic, oom, index out of bounds etc.

Reply:
Okay so the only way this might fail is if we panic? If so, maybe do a
```
FuzzParseProwConfig(data []byte) int} (err error) {
    defer func() {
         if r := recover(); r != nil {
            err = fmt.Errorf("found a panic: %v", r)
         }
     }
}
```

We btw have a bunch of fuzztests already that run as part of the normal `go test`: https://grep.app/search?q=gofuzz%22&filter[repo][0]=kubernetes/test-infra